### ISSUE TITLE
[MNG-6939] ITs fail when MAVENCODEBASE is relative

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8101]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[MNG-6939](https://issues.apache.org/jira/browse/MNG-6939?redirect=false)** and commented

Reproducer:

```
osipovmi@deblndw011x:~/var/Projekte/maven-integration-testing
$ export MAVENCODEBASE=../maven
$ ./run-its.sh
...
[INFO] Maven Core ITs suite 2.1-SNAPSHOT .................. FAILURE [  5.221 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 57.130 s
[INFO] Finished at: 2020-06-10T11:09:06+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.0.0:run (unpack-mav                                                                                                                  en-distro) on project core-it-suite: An Ant BuildException has occured: src '/var/osipovmi/Projek                                                                                                                  te/maven-integration-testing/maven/apache-maven/target/apache-maven-bin.zip' doesn't exist.
[ERROR] around Ant part ...<unzip src="../maven/apache-maven/target/apache-maven-bin.zip" dest="/                                                                                                                  var/osipovmi/Projekte/maven-integration-testing/core-it-suite/target/apache-maven">... @ 5:158 in                                                                                                                   /var/osipovmi/Projekte/maven-integration-testing/core-it-suite/target/antrun/build-main.xml
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following a                                                                                                                  rticles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :core-it-suite
```

It is apparently not obvious that the env var is is normalized, nested or cannot be relative.

[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte), [Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy), can you both have a look? We either should fail properly or support it.

---

**Remote Links:**
- [GitHub Pull Request #65
](https://github.com/apache/maven-integration-testing/pull/65)

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

I'll add a fix for `cmd`.
While at it, let's add another improvement in this ticket:
In case the build of maven core fails, it should exit the script.
Again, I'll fix it for `cmd`, that's something I can verify.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Thanks, please add your commits to the already existing branch. We can then verify both.
I would prefer that the normalization would happen in the POM, but did not actually found a decent way for ths.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with ab987368b8ec5148f5b17c817c6b3d7e93a083f9 and 2002bac42a90f7702b83c1c54c46b169c6ca0f55.
